### PR TITLE
fix: Prettier major update

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/types.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export interface PostUpgradeTasks {
 }
 
 export type UpdateConfig<
-  T extends RenovateSharedConfig = RenovateSharedConfig
+  T extends RenovateSharedConfig = RenovateSharedConfig,
```
Comment (viceice) [inline lib/config/types.ts:179]: we can do the trailing comma change separate via prettier config, so this PR gets much smaller.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -344,6 +344,8 @@ jobs:
 
       - name: Unit tests
         shell: bash
+        env:
+          NODE_OPTIONS: --experimental-vm-modules
```
Comment (viceice) [inline .github/workflows/build.yml:348]: why do we need this?

Comment (rarkins) [reply .github/workflows/build.yml:348]: Presumably fixing https://github.com/renovatebot/renovate/pull/23627#issuecomment-1658328852

**Review by rarkins — APPROVED** [review]

**Comment by rarkins** [comment]
Based on the commit history, this looks clean to me:
- Update prettier to v3
- Update build workflow to fix failing unit test
- Run prettier-fix

@viceice do you have any better way to fix the unit test failures?

**Comment by viceice** [comment]
first create a PR and change this to `all`

https://github.com/renovatebot/renovate/blob/6d48407d1365f795995a726a6881cbc35f3f6767/.prettierrc#L2

this works with prettier v2 and v3. the es5 mode was required for very ancient Browsers like Internet Explorer.

then do all other stuff to fix here.

**Comment by rarkins** [comment]
> first create a PR and change this to `all`

I thought we had it this way because we didn't like the all style. I tried this locally and it changes 1123 files versus 51 in this PR. I also don't like these changes. Are you sure we need to do this?

**Comment by viceice** [comment]
> > first create a PR and change this to `all`
> 
> I thought we had it this way because we didn't like the all style. I tried this locally and it changes 1123 files versus 51 in this PR. I also don't like these changes. Are you sure we need to do this?
> 

🤔 I thought all those comma changes are because of that setting. I've set the all setting on almost all other repos where I'm migrating to prettier v3. I like the trailing comma because they cause less conflicts on later changes.

**Comment by rarkins** [comment]
It's lots of changes like this:

image

File: .github/workflows/build.yml (hunk 2 above)
Comment (viceice) [reply .github/workflows/build.yml:348]: ah, right. prettier is esm and has a simple cus wrapper with dynamic import.

**Review by viceice — APPROVED** [review]

**Pull request merged**